### PR TITLE
Updated containers with organized comments.

### PR DESCRIPTION
**Pull request by mstraut**

### CODE REVIEW

**Review by mstraut — COMMENTED** [review]
all comments are now placed properly

**Review by mstraut — COMMENTED** [review]
moved some comments, and updated import

File: src/prog_models/utils/containers.py [hunk 1]
```diff
@@ -4,24 +4,23 @@
...
-        The contained data (e.g., :term:`input`, :term:`state`, :term:`output`). If numpy array should be column vector in same order as keys
     """
-    def __init__(self, keys : list, data : Union[dict, np.array]):
-        if not isinstance(keys, list):
-            keys = list(keys)
 
+    def __init__(self, keys: list, data: Union[dict, np.array]):
+        """ Initializes the container
+
+        Arguments:
```
Comment (teubert) [inline src/prog_models/utils/containers.py:18]: For the sphinx doc generation to work correctly, the arguments for the constructor need to be in the doc comments for the class. You can repeat it here if you feel it helps, but add it back to the doc comment above. 

Also, note that doc comments need to be in numpy or google format for sphinx doc generation to work. 

* Numpy: https://numpydoc.readthedocs.io/en/latest/format.html or https://python.plainenglish.io/how-to-write-numpy-style-docstrings-a092121403ba
* Google: https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

**Review by teubert — CHANGES_REQUESTED** [review]
All these comments are helpful. Once the constructor doc comment is fixed I'm ready to approve. 

Thanks!

File: src/prog_models/utils/containers.py [hunk 2]
```diff
@@ -35,79 +34,147 @@ def __init__(self, keys : list, data : Union[dict, np.array]):
...
     def values(self) -> np.array:
-        if len(self.matrix) > 0 and len(self.matrix[0]) == 1:
-            return np.array([value[0] for value in self.matrix])
-        return self.matrix
+        """
+        returns array of matrix values
+        """
+        if len(self.matrix) > 0 and len(self.matrix[0]) == 1:   #if the first row of the matrix has one value (i.e., non-vectorized)
+            return np.array([value[0] for value in self.matrix])    #the value from the first row
+        return self.matrix_check    #he matrix (vectorized case)
```
Comment (teubert) [inline src/prog_models/utils/containers.py:137]:
```suggestion
        return self.matrix    # the matrix (vectorized case)
```

I think this is supposed to be self.matrix. I think this is why the tests are failing

**Review by teubert — APPROVED** [review]
Can be merged after the typo in values is fixed and tests pass.

Great improvements to documentation and readability of containers

**Pull request merged**